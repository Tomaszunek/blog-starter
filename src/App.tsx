import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import './App.css';


export namespace App {
  export interface IProps extends RouteComponentProps<void> {
    numbers: any   
  }
}

export default class App extends React.Component<App.IProps> {    
  public render() {
    return (
      <div className="App">
        abc
      </div>
    );
  }
}


