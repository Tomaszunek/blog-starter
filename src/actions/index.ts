export * from './articles';
export * from './products';
export * from './mainpage';
export * from './abilities';
export * from './projects';
export * from './appSystem';

