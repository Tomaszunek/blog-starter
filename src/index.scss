@import 'src/styles/variables.scss';

body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
  background-color: $body-background;
}

.container {
  padding: 0px 100px;
  background-color: $primary-background;

  @media only screen and (min-width: $sm) and (max-width: $lg) {
    padding: 0px 80px;
  }
  @media only screen and (min-width: $xs) and (max-width: $sm) {
    padding: 0px 60px;
  }
  @media only screen and (max-width: $xs) {
    padding: 0px 40px;ab
  }   
}