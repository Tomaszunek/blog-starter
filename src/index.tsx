import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Application } from './router';
import { configureStore } from './store';
import { createBrowserHistory } from 'history';
import registerServiceWorker from './registerServiceWorker';
import { Provider } from 'react-redux';
import { Router, Switch } from 'react-router';
import Navigation from 'src/pages/Navigation';
import Footer from 'src/pages/Footer';
import './index.scss';

const store = configureStore();
const history = createBrowserHistory();

ReactDOM.render(
  <Provider store={store}>    
    <Router history={history}>
      <div>
        <Navigation/>
          <Switch>
            <Application />
          </Switch>
        <Footer/>
      </div>
    </Router>   
  </Provider>,  
  document.getElementById('root') as HTMLElement
);
registerServiceWorker();