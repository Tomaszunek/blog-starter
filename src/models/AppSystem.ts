export interface IAppSystem {
    loading: boolean   
} 