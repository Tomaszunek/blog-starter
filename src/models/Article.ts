

export interface IArticle {
    id: number,
    name: string,
    body: string
}
    
  
export namespace ArticleFiltes {
    export enum Filter {
        SHOW_ALL = 'all',
        SHOW_ACTIVE = 'active',
        SHOW_COMPLETED = 'completed'
    }
}