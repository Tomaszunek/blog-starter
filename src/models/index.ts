export * from './ArticleModel';
export * from './ProductModel';
export * from './ContentCategory';
export * from './MainPageContentModel';
export * from './AppSystem';

