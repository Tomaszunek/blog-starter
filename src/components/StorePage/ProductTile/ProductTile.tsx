import * as React from 'react';
import './ProductTile.scss'
import { IProductModel } from '../../../models';
import CategoryTags from '../../template/categoryTags/categoryTags';

export default class ProductTile extends React.Component<IProductTilesProps, any> {
  constructor(props: IProductTilesProps) {
    super(props);    
  }

  public render() {         
    const { id, image, name, AppContentCategories, slug, type } = this.props.product;
    return (
    <div className="articleTile" data-attr={id}>
        <a href={`/product/${slug}`}>           
        <img src={'./images/placeholders/' + image} alt=""/>
        <div className="tileCont">
        {
            AppContentCategories && AppContentCategories.length ? 
            <CategoryTags categories={AppContentCategories} articleType={type}/>  :
            null        
        }
            <div>
            <span>{name}</span>
            {
                (this.props.product ? <p>16.99$</p> : null)
            }              
            </div>
        </div>
        </a>       
    </div>
    );    
  }
}

export interface IProductTilesProps{
    product: IProductModel,
}


