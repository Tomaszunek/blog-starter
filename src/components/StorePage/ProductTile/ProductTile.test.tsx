import * as React from 'react';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';
import ProductTile from './ProductTile'
import { IProductModel } from 'src/models';

describe('ProjectPage', () => {
  describe('Component should render right', () => {
    let app:any;    
    const product:IProductModel ={
        id: 1,
        type: "ability",
        body: "body",
        createdAt: new Date(1545167325 *1000),
        image: "images/placeholder/abc.jpg",
        name: "woda",
        slug: "woda",
        updatedAt: new Date(1545167325 *1000),
        productType: "type", 
        AppContentCategories: [
            {id: 1, name: "category", type: "type"}, 
            {id: 2, name: "category", type: "type"}
        ]
    }        
    beforeEach(() => {
      app = mount( 
        <ProductTile product={product}/>
      )
    })
    it('render correctly snapshot component', () => {
      const tree = renderer
      .create(<ProductTile product={product}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
    })    
    it('should contain project containar', () => {      
        expect(app.find('.articleTile').length).toEqual(1)               
    })
    it('should contain elements', () => {
        expect(app.find('img').length).toEqual(1)
        expect(app.find('.categoryTagsCont').length).toEqual(1)
        expect(app.find('.category').length).toEqual(2)
        expect(app.find('a[href="/product/woda"]').length).toEqual(1)
    })
    it('should have proper props', () => {
      expect(app.instance().props.product).toEqual(product)
    })   
  });
}); 