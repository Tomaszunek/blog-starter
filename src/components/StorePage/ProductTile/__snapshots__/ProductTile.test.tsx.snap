// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectPage Component should render right render correctly snapshot component 1`] = `
<div
  className="articleTile"
  data-attr={1}
>
  <a
    href="/product/woda"
  >
    <img
      alt=""
      src="./images/placeholders/images/placeholder/abc.jpg"
    />
    <div
      className="tileCont"
    >
      <div
        className="categoryTagsCont ability"
      >
        <p
          className="category"
        >
          a
        </p>
        <p
          className="category"
        >
          a
        </p>
      </div>
      <div>
        <span>
          woda
        </span>
        <p>
          16.99$
        </p>
      </div>
    </div>
  </a>
</div>
`;
